samtranslator/model/api/api_generator.py

```diff
@@ -1128,6 +1128,15 @@ def _openapi_postprocess(self, definition_body: Dict[str, Any]) -> Dict[str, Any
         if definition_body.get("openapi") is not None and self.open_api_version is None:
             self.open_api_version = definition_body.get("openapi")
 
+        if (
+            definition_body.get("openapi") is not None
+            and self.open_api_version is not None
+            and definition_body.get("openapi") != self.open_api_version
```

Comment:
Checking the major version might be better as major versions should allow for the template to be parsed properly. This would make it so `3.0.1` and `3.0.0` would not match and that might be too strict for this use case.

Reply:
Just decided to use a different approach to address the issue.